src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -1171,9 +1171,10 @@ private BinaryOperatorAnalysisResult BinaryOperatorOverloadResolution(BinaryOper
                 resultKind = possiblyBest.HasValue ? LookupResultKind.Viable : LookupResultKind.Empty;
             }
 
-            if (possiblyBest.HasValue)
+            if (possiblyBest is { HasValue: true, Signature: { Method: { } bestMethod } })
             {
-                ReportObsoleteAndFeatureAvailabilityDiagnostics(possiblyBest.Signature.Method, node, diagnostics);
+                ReportObsoleteAndFeatureAvailabilityDiagnostics(bestMethod, node, diagnostics);
+                ReportUseSiteDiagnostics(bestMethod, diagnostics, node);
```

Comment:
Just to confirm, this change is necessary for `UnmanagedCallersOnlyDefinedOnBinaryOperator_InMetadata` case, right?

Reply:
Correct.

---
In reply to: [483177586](https://github.com/dotnet/roslyn/pull/47311#discussion_r483177586) [](ancestors = 483177586)